pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py

```diff
@@ -65,13 +65,23 @@ def get_waveform(self):
         self.write("C{ch}:WF? ALL")
         response = self.read_bytes(count=-1, break_on_termchar=True)
         descriptor_dictionary = self.get_descriptor()
-        rawWaveform = list(
-            struct.unpack_from(
-                "%db" % descriptor_dictionary["numDataPoints"],
-                response,
-                offset=descriptor_dictionary["descriptorOffset"],
-            ),
-        )
+        if descriptor_dictionary["numDataPoints"] == 0:
+            rawWaveform = list(
+                struct.unpack_from(
+                    "%db" % descriptor_dictionary["pointsScreen"],
+                    response,
+                    offset=descriptor_dictionary["descriptorOffset"],
```

Comment:
If all data points are returned, should not the offset be 0?

Reply:
The first bytes are a descriptor string and do not contain data. This is why they are skipped.